### PR TITLE
Updating doc strings for evaluate API

### PR DESCRIPTION
**Pull request by singankit**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by nagkumar91 — APPROVED** [review]

**Review by luigiw — DISMISSED** [review]

**Review by nagkumar91 — DISMISSED** [review]

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 1]
```diff
@@ -283,21 +284,30 @@ def evaluate(
...
+          the results will be saved to a file named `evaluation_results.json` in the folder.
     :paramtype output_path: Optional[str]
-    :keyword azure_ai_project: Logs evaluation results to AI Studio
+    :keyword azure_ai_project: Logs evaluation results to AI Studio if set.
     :paramtype azure_ai_project: Optional[Dict]
-    :return: A EvaluationResult object.
-    :rtype: ~azure.ai.generative.evaluate.EvaluationResult
+    :return: Evaluation results.
+    :rtype: dict
     """
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:302]: This is the most important API. Shall we add some sample code to docstring?

**Review by luigiw — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 2]
```diff
@@ -283,21 +284,64 @@ def evaluate(
...
+    :rtype: dict
+
+    :Example:
+
+    Evaluate API can be used as follows:
+
+    .. code-block:: python
+
+            from promptflow.evals.evaluate import evaluate
+            from promptflow.evals.evaluators import RelevanceEvaluator, CohereEvaluator
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:310]: typo: CohereEvaluator

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 3]
```diff
@@ -283,21 +284,64 @@ def evaluate(
...
+
+            coherence_eval = CohereEvaluator()
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:312]: same here

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 4]
```diff
@@ -283,21 +284,64 @@ def evaluate(
...
+            path = "evaluate_test_data.jsonl"
+            result = evaluate(
+                data=path,
+                evaluators={
+                    "coherence": coherence_eval,
+                    "relevance": relevance_eval,
+                },
+                evaluator_config={
+                    "coherence": {
+                        "answer": "data.answer",
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:324]: shouldn't be ${data.answer}

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py (hunk 3 above)
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:312]: Add ... to the constructor?
coherence_eval = CohereEvaluator(...)

**Review by nagkumar91 — DISMISSED** [review]

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py (hunk 4 above)
Comment (singankit) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py]: Thanks for catching that. Have fixed it

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py (hunk 3 above)
Comment (singankit) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py]: Thanks for catching that. Have fixed it

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 5]
```diff
@@ -283,21 +284,72 @@ def evaluate(
...
+            result = evaluate(
+                data=path,
+                evaluators={
+                    "coherence": coherence_eval,
+                    "relevance": relevance_eval,
+                },
+                evaluator_config={
+                    "coherence": {
+                        "answer": "${data.answer}",
+                        "context": "${data.context}",
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:333]: Coherence doesn't have context as input.

**Review by ninghu — APPROVED** [review]

**Review by nagkumar91 — APPROVED** [review]

**Pull request merged**